src/engine_redraw.c

```diff
@@ -99,15 +99,37 @@ static void draw_creature_view_icons(struct Thing* creatng)
         y = MyScreenHeight - scale_ui_value_lofi(spr->SHeight * 2);
     }
     struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
-    for (int Spell = SplK_Freeze; Spell < SplK_TimeBomb; Spell++)
+    for (SpellKind Spell = SplK_Freeze; Spell <= SplK_TimeBomb; Spell++)
     {
         if (creature_affected_by_spell(creatng, Spell))
         {
             struct SpellConfig* spconf = get_spell_config(Spell);
             long spridx = spconf->medsym_sprite_idx;
-            if ( (Spell == SplK_Invisibility) && (cctrl->force_visible & 2) )
+            if (Spell == SplK_Invisibility)
             {
-                spridx++;
+                if (cctrl->force_visible & 2)
+                {
+                    spridx++;
+                }
+            }
+            else if (Spell == SplK_TimeBomb)
```

Comment:
why does timebomb need special code here?

Reply:
For the countdown.